cmd/snap/cmd_debug_timings.go

```diff
@@ -126,15 +105,99 @@ func (x *cmdChangeTimings) Execute(args []string) error {
 			fmt.Fprintf(w, "%s\t%s\t%11s\t%11s\t%s\n", t.ID, t.Status, doingTime, undoingTime, summary)
 		}
 
-		for _, nested := range timings[t.ID].DoingTimings {
+		for _, nested := range timing.ChangeTimings[t.ID].DoingTimings {
 			showDoing := true
 			printTiming(w, &nested, x.Verbose, showDoing)
 		}
-		for _, nested := range timings[t.ID].UndoingTimings {
+		for _, nested := range timing.ChangeTimings[t.ID].UndoingTimings {
 			showDoing := false
 			printTiming(w, &nested, x.Verbose, showDoing)
 		}
 	}
+
+	return nil
+}
+
+func (x *cmdChangeTimings) printEnsureTimings(w io.Writer, timings []*timingsData) error {
+	for _, td := range timings {
+		if len(td.EnsureTimings) > 0 {
+			for _, t := range td.EnsureTimings {
+				if x.Verbose {
+					fmt.Fprintf(w, "%s\t%s\t%11s\t%11s\t%s\t%s\n", "ensure", "-", formatDuration(t.Duration), "-", t.Label, t.Summary)
```

Comment:
I am a bit confused, what do we do with level/nesting here, the timings from Ensure itself can be nested, no?

Reply:
As discussed, I forgot about nesting indicator here, thanks for catching.

It's now added, plus some refactoring to have one printing helper for all timings. Note though, we won't see any nesting for current ensure timings (unless I missed something) as are all level 0; for the "refresh-catalogs" ensure discussed on IRC, we have "refresh-catalogs" root timing node (with tags), and two level-0 measurements under it. So the output is:
```
pawel@ubuntu:~/gocode/src/github.com/snapcore/snapd/cmd/snap$ ./snap debug timings --ensure=refresh-catalogs --all --verbose
ID      Status        Doing      Undoing  Label           Summary
ensure                 34ms            -  get-sections    query store for sections
ensure               1835ms            -  write-catalogs  query store for catalogs
ensure                209ms            -  get-sections    query store for sections
ensure                374ms            -  write-catalogs  query store for catalogs
```

Reply:
yes, as we discussed the latter seems maybe a good idea (for clarity mostly),  but as we also discussed can be a follow up